src/GitVersion.Core.Tests/VersionCalculation/Strategies/ConfigNextVersionBaseVersionStrategyTests.cs

```diff
@@ -48,8 +48,8 @@ public void ConfigNextVersionTest(string nextVersion, string expectedVersion)
         var strategy = contextBuilder.ServicesProvider.GetServiceForType<IVersionStrategy, ConfigNextVersionVersionStrategy>();
         var context = contextBuilder.ServicesProvider.GetRequiredService<Lazy<GitVersionContext>>().Value;
         var branchMock = GitToolsTestingExtensions.CreateMockBranch("main", GitToolsTestingExtensions.CreateMockCommit());
-        var branchConfiguration = context.FullConfiguration.GetBranchConfiguration(branchMock);
-        var effectiveConfiguration = new EffectiveConfiguration(context.FullConfiguration, branchConfiguration);
+        var branchConfiguration = context.Configuration.GetBranchConfiguration(branchMock);
```

Comment:
What's the difference between `Config.GetBranchConfiguration()` and `Config.ForBranch()` and why aren't we using the latter here?

```suggestion
        var branchConfiguration = context.Configuration.ForBranch(branchMock);
```

Reply:
Good question. GetBranchConfiguration is the new implementation which ensures that always a branch will be returned (no null values are allowed). If the user has no branch configuration defined the unknown branch configuration will be returned. GetBranchConfiguration  uses ForBranch internally.

Reply:
Okay. Not something we need to tackle here, but I think we should only have one public method that returns the configuration for a specific branch.

Reply:
Like you already mentioned I think we need to consolidate the ConfigExtensions class and place the logic somewhere else. Another PR? ;)

Reply:
Yep, absolutely!